app/core/memory/memory_service.py

```
@@ -0,0 +1,82 @@
+        self,
+        agent_id: str,
+        user_id: str,
+        message_type: str,
+    ) -> KarakuriMemory:
+        try:
+            return await zep_client.get_memory(
+                session_id=self._create_session_id(agent_id, user_id, message_type),
+                lastn=30,
+            )
+        except Exception:
+            raise HTTPException(status_code=500, detail="Failed to get session memory")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)